src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -83,12 +87,48 @@ public class PPLTool implements Tool {
 
     private String contextPrompt;
 
+    private PPLModelType pplModelType;
+
     private static Gson gson = new Gson();
 
-    public PPLTool(Client client, String modelId, String contextPrompt) {
+    private static Map<String, String> defaultPromptDict;
+
+    static {
+        try {
+            defaultPromptDict = loadDefaultPromptDict();
+        } catch (IOException e) {
+            log.error("fail to load default prompt dict" + e.getMessage());
+            defaultPromptDict = new HashMap<>();
+        }
+    }
+
+    public enum PPLModelType {
+        CLAUDE,
+        FINETUNE;
+
+        public static PPLModelType from(String value) {
+            if (value.isEmpty()) {
```

Comment:
Is value possibly be null?

Reply:
No, when create ppltool object, we use `(String) map.getOrDefault("model_type", "")` So model type string will have default value.